skopt/tests/test_space.py

```diff
@@ -84,6 +84,22 @@ def test_categorical_transform():
     assert_array_equal(ent_inverse, categories)
 
 
+def test_categorical_transform_binary():
+    categories = ["apple", "orange"]
+    cat = Categorical(categories)
+
+    apple = [0.]
+    orange = [1.]
+
+    assert_equal(cat.transform(["apple"]).size, cat.transformed_size)
```

Comment:
ah, yes that makes sense. I've added it. went ahead and added the same size check for the non-binary case